open_optical_gating/app/app.py

```diff
@@ -330,4 +340,18 @@ def run():
     return render_template("emulate.html")
 
 if __name__ == "__main__":
+    # TODO: JT writes: I am still uncertain whether we are talking about concurrent multithreading
+    # or just sequential here. I am actually now thinking that *flask* is (or at least could potentially)
+    # spawn multiple *processes* to serve these requests, i.e. genuine concurrent multiprocessing.
+    # My blunt question is: has anybody writing this thought through the implications of this?
+    # How does that work in terms of state, race conditions etc? I'm wondering now if the "global analyse_camera"
+    # line (and associated comment) was written by you guys or was in an example you have cloned.
+    # I think this needs some careful thought, understanding and documenting, because I think there's
+    # a risk of things getting very messy. At the very least, we need to be very clear about
+    # how the threading is behaving.
+    # This is where my earlier observation to Chas becomes very important, making the point that
+    # this code is not implementing true REST because this code is not *stateless*.
+    # That's no longer just pedantry when we are talking about concurrent multithreading,
+    # it's really important!
+    
```

Comment:
I'm hoping to avoid learning too much of the internals of flask myself, but if you know of an easy way to implement cookies, that might(?) be a pre-packaged way to monitor for (and reject) multiple connections, if implemented with a little bit of care?

Reply:
Yes, using session (like cookies but server side) can negate assumption 1 (https://pythonbasics.org/flask-sessions/) if we set an ID for the user then we could stop sessions interacting.